mono/tests/Makefile.am

```diff
@@ -1507,6 +1507,19 @@ sgen-bridge3-tests-plain-tarjan-bridge: $(SGEN_BRIDGE3_TESTS) test-runner.exe
 sgen-bridge3-tests-ms-split-tarjan-bridge: $(SGEN_BRIDGE3_TESTS) test-runner.exe
 	MONO_ENV_OPTIONS="--gc=sgen" MONO_GC_DEBUG="bridge=3Bridge" MONO_GC_PARAMS="bridge-implementation=tarjan,minor=split" $(TOOLS_RUNTIME) $(TEST_RUNNER) --testsuite-name $@ --timeout 900 $(SGEN_BRIDGE3_TESTS)
 
+SGEN_OOM_TESTS_SRC =	\
+	sgen-oom-cementing-stress.cs	\
+	sgen-oom-new-threads-dont-join-stw-2.cs	\
+	sgen-oom-bridge-major-fragmentation.cs \
+	gc-oom-handling.exe	\
+	gc-oom-handling2.exe
+
+SGEN_OOM_TESTS=$(SGEN_OOM_TESTS_SRC:.cs=.exe)
+
+# If limiting test-runner memory is problematic, we should add argument for env passing
+sgen-oom-tests: $(SGEN_OOM_TESTS) test-runner.exe sgen-cementing-stress.exe sgen-new-threads-dont-join-stw-2.exe sgen-bridge-major-fragmentation.exe
+	MONO_ENV_OPTIONS="--gc=sgen" MONO_GC_DEBUG="max-valloc-size=50M,bridge=Bridge" $(TOOLS_RUNTIME) $(TEST_RUNNER) $(TEST_RUNNER_ARGS) --testsuite-name $@ --disabled "$(DISABLED_TESTS)" --timeout 900 $(SGEN_OOM_TESTS)
```

Comment:
test-runner already supports `--runtime-args` so you could just pass `--runtime-args="--gc=sgen --gc-debug=max-valloc-size=50M,bridge=Bridge"` instead of setting MONO_ENV_OPTIONS MONO_GC_DEBUG so those settings will only get passed to the test process.

We should probably change the existing tests too.